### PR TITLE
Issue 65 hamster combo 3 of a kind

### LINKED ISSUE #65: Hamster Combo: 3-of-a-Kind — Take Specific Card
Playing 3 identical Hamster cards takes a named card from a target player.

Acceptance Criteria:
  - Same logic as Beg for Snacks — reuse it
  - Tests: all 5 Hamster types

  Depends on: #62

### PR DESCRIPTION
**Pull request by DrCockblock**
logic for the hamster trio

### CODE REVIEW

**Comment by OlgaKupar** [comment]
Please keep the handsize = 16 even if it fails for now (should be 15 to pass at the current state), because after we merge PR #92 , the check will be successful again)

File: src/main/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinition.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  public HamsterTrioCardDefinition() {}
+
+  @Override
+  public String cardType() {
+    return "HamsterTrio";
+  }
+
+  @Override
+  public String commandId() {
+    return "HamsterTrio";
```
Comment (OlgaKupar) [inline src/main/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinition.java]: please adapt it to our internal convention: 

HamsterTrio -> HAMSTER_TRIO

File: src/main/java/com/doomhamsters/gamesession/cardcommands/CardRegistry.java [hunk 2]
```diff
@@ -65,6 +66,9 @@ public static CardRegistry defaultRegistry() {
         new SniffAheadCardDefinition(),
         new SignOfFateCardDefinition(),
         new StealCardDefinition(),
+        new BegForSnacksCardDefinition(),
```
Comment (OlgaKupar) [inline src/main/java/com/doomhamsters/gamesession/cardcommands/CardRegistry.java]: remove duplicate line

*might be caused during merge conflict solving by me, if so - sorry in advance😉

File: src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java
Comment (OlgaKupar) [inline src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java:1]: Overall, very good tests. 
I´d have ideas, how to make it a bit more comprehencive. You could test: 
- self-targeting
- not enough hamster cards in the deck
- target has no matching card
- missing targetPlayerId or missing cardType
- full success path (including the card transfer)

**Review by OlgaKupar — CHANGES_REQUESTED** [review]
Well done! 
Please take a look at review comments - would you mind take this suggestions and make changes? thanks=)

**Review by SereinigChristian — COMMENTED** [review]
The core logic for the Hamster Trio looks really solid. 

I'm holding off on approving for now, as I completely agree with @OlgaKupar's review. Let's get the naming convention for the commandId aligned, drop that duplicate line in the registry, and round out the test suite with the edge cases she mentioned.

Once those updates are pushed, ping me and I'll be happy to approve!

File: src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java [hunk 3]
```diff
@@ -95,4 +100,70 @@ void execute_throwsWhenRequiredParameterMissing() {
...
+    CardCommandContext context =
+      createContext(
+        Map.of(
+          "targetPlayerId", target.getId(),
+          "cardType", "beg_for_snacks",
+          "hamsterType", "hamster_ninja"));
+
+    CardCommandResult result = definition.execute(context);
+
+    assertFalse(requester.getHand().contains(wanted));
```
Comment (OlgaKupar) [inline src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java]: Reversed: we want to check that the card is in requester´s hand now; 

must be assert**True**

File: src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java [hunk 4]
```diff
@@ -95,4 +100,70 @@ void execute_throwsWhenRequiredParameterMissing() {
...
+    assertTrue(target.getHand().contains(wanted));
```
Comment (OlgaKupar) [inline src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java]: Reversed: the card is now gone from target; 

must be assert**False**

File: src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java [hunk 5]
```diff
@@ -95,4 +100,70 @@ void execute_throwsWhenRequiredParameterMissing() {
...
+    assertEquals(0, requester.getHand().size()); // only stolen card remains
```
Comment (OlgaKupar) [inline src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java]: that is true for target.getHand().size())

but here if you want to check the requestors´s hand, you need: 

assertEquals(1, requester.getHand().size()) - 1 here because 3 hamster cards were discarded and 1 card got stollen from the target

File: src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java [hunk 6]
```diff
@@ -95,4 +100,70 @@ void execute_throwsWhenRequiredParameterMissing() {
...
+    assertFalse(result.getPublicMessage().contains("took"));
```
Comment (OlgaKupar) [inline src/test/java/com/doomhamsters/gamesession/cardcommands/cards/HamsterTrioCardDefinitionTest.java]: Reversed: 

should be assertTrue - we indeed want to get a success message here

**Review by OlgaKupar — DISMISSED** [review]
Assertions are reversed - please see comments. 
You could also add @ParameterizedTest across 5 hamsters types (see TwoHamstersCardDefinitionTest).

**Review by TheRealSlyzzz — APPROVED** [review]
After some difficulties, I'm more than happy to approve the changes for merge. :)

**Review by michelle-fe — APPROVED** [review]
Solid implementation — great KDoc, commandId naming is aligned now, and the edge cases (self-target, eliminated target, too few hamsters, no match) are all handled. Nice work
